tests/Functional/src/com/microsoft/aad/adal/test/AuthenticationContextTest.java

```diff
@@ -1875,6 +1873,85 @@ public void testBrokerRedirectUri() throws UnsupportedEncodingException {
         assertTrue("should have signature url encoded",
                 actual.contains(URLEncoder.encode(testTag, AuthenticationConstants.ENCODING_UTF8)));
     }
+    
+    @SmallTest
+    public void testVerifyManifestPermission() throws NoSuchAlgorithmException, NoSuchPaddingException, IllegalArgumentException, ClassNotFoundException, NoSuchMethodException, InstantiationException, IllegalAccessException, InvocationTargetException{
+        //mock context
+        FileMockContext mockContext = new FileMockContext(getContext());
+        Boolean actual;
+        
+        //set the permission
+        mockContext.setPermission("android.permission.INTERNET",PackageManager.PERMISSION_GRANTED);
+        mockContext.setPermission("android.permission.GET_ACCOUNTS",PackageManager.PERMISSION_GRANTED);
+        mockContext.setPermission("android.permission.MANAGE_ACCOUNTS",PackageManager.PERMISSION_GRANTED);
+        mockContext.setPermission("android.permission.USE_CREDENTIALS",PackageManager.PERMISSION_GRANTED);
+        AuthenticationContext context = new AuthenticationContext(mockContext, VALID_AUTHORITY, false);
+
+        //set the access to private method
+        Method m = ReflectionUtils.getTestMethod(context, "verifyManifestPermissions");
+        
+        //test@case all permissions are granted
+        actual = (Boolean)m.invoke(context);
+        assertTrue(actual);
+        
+        //test@case permission GET_ACCOUNTS is missing
+        try
+        {
+            mockContext.removePermission("android.permission.GET_ACCOUNTS");
+            m.invoke(context);
+            Assert.fail("It is expected to return an exception here.");
+        }
+        catch(InvocationTargetException e)
+        {
+            assertTrue(e.getCause() instanceof AuthenticationException);
+            assertTrue(((AuthenticationException)e.getCause()).getMessage().toString().contains("[GET_ACCOUNTS]"));
+            assertEquals(ADALError.DEVELOPER_BROKER_PERMISSIONS_MISSING,((AuthenticationException)e.getCause()).getCode());
+        }
+        
+        //test@case permission MANAGE_ACCOUNTS is missing
```

Comment:
I like to have different tests for individual test case, there may be duplicate, but it's so clear what every single test is doing.